### PR TITLE
Misuv 8738

### PR DESCRIPTION
**Pull request by WozCapGem**

### CODE REVIEW

File: test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 1]
```diff
@@ -195,7 +195,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport {
 
     def changeReportingMethodUrl(id: String, taxYear: String, changeTo: String): String = {
       controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController
-        .show(taxYear, changeTo, incomeSourceType = SelfEmployment, isAgent = isAgent).url
+        .show(taxYear, changeTo, incomeSourceType = SelfEmployment).url
```
Comment (Chrissystk) [inline test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala:198]: Again this should be an if statement

File: test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 2]
```diff
@@ -244,7 +244,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport {
   class ukSetupUnknowns(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, UkProperty).url
+      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, UkProperty).url
```
Comment (Chrissystk) [inline test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala:247]: Again this should be an if statement

File: test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 3]
```diff
@@ -261,7 +261,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport {
   class foreignSetup(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, ForeignProperty).url
+      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, ForeignProperty).url
```
Comment (Chrissystk) [inline test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala:264]: Again this should be an if statement

File: test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 4]
```diff
@@ -278,7 +278,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport {
   class foreignSetupUnknowns(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, ForeignProperty).url
+      controllers.incomeSources.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, ForeignProperty).url
```
Comment (Chrissystk) [inline test/views/incomeSources/manage/ManageIncomeSourceDetailsViewSpec.scala:281]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 5]
```diff
@@ -345,7 +345,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
 
     def changeReportingMethodUrl(id: String, taxYear: String, changeTo: String): String = {
       controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController
-        .show(taxYear, changeTo, incomeSourceType = SelfEmployment, isAgent = isAgent).url
+        .show(taxYear, changeTo, incomeSourceType = SelfEmployment).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:348]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 6]
```diff
@@ -364,7 +364,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
...
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:367]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 7]
```diff
@@ -417,7 +417,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
   class ukSetup(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, UkProperty).url
+      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo,UkProperty).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:420]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 8]
```diff
@@ -384,7 +384,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
 
     def changeReportingMethodUrl(id: String, taxYear: String, changeTo: String): String = {
       controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController
-        .show(taxYear, changeTo, incomeSourceType = SelfEmployment, isAgent = isAgent).url
+        .show(taxYear, changeTo, incomeSourceType = SelfEmployment).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:387]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 9]
```diff
@@ -452,7 +452,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
...
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:455]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 10]
```diff
@@ -485,7 +485,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
   class foreignSetup(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, ForeignProperty).url
+      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, ForeignProperty).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:488]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 11]
```diff
@@ -502,7 +502,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
   class foreignSetupUnknowns(isAgent: Boolean, error: Boolean = false) {
 
     def changeReportingMethodUrl(taxYear: String, changeTo: String): String = {
-      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, isAgent, ForeignProperty).url
+      controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController.show(taxYear, changeTo, ForeignProperty).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:505]: Again this should be an if statement

File: test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala [hunk 12]
```diff
@@ -520,7 +520,7 @@ class ManageIncomeSourceDetailsViewSpec extends TestSupport with ViewSpec {
 
     def changeReportingMethodUrl(id: String, taxYear: String, changeTo: String): String = {
       controllers.manageBusinesses.manage.routes.ConfirmReportingMethodSharedController
-        .show(taxYear, changeTo, incomeSourceType = SelfEmployment, isAgent = isAgent).url
+        .show(taxYear, changeTo, incomeSourceType = SelfEmployment).url
```
Comment (Chrissystk) [inline test/views/manageBusinesses/manage/ManageIncomeSourceDetailsViewSpec.scala:523]: Again this should be an if statement

**Review by Chrissystk — CHANGES_REQUESTED** [review]
Have the ATs run and passed?

File: app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala [hunk 13]
```diff
@@ -272,9 +293,9 @@ user.incomeSources.getLatencyDetails(incomeSourceType, id.value) match {
     (backCall, successCall)
   }
 
-  private def getPostAction(taxYear: String, changeTo: String, isAgent: Boolean, incomeSourceType: IncomeSourceType): Call = {
+  private def getPostAction(taxYear: String, changeTo: String, incomeSourceType: IncomeSourceType): Call = {
     routes.ConfirmReportingMethodSharedController
-      .submit(taxYear, changeTo, isAgent, incomeSourceType)
+      .submit(taxYear, changeTo, incomeSourceType)
```
Comment (Chrissystk) [inline app/controllers/incomeSources/manage/ConfirmReportingMethodSharedController.scala:298]: This is the cause of your error. You need the submitAgent when the user is an agent.

File: app/controllers/manageBusinesses/manage/ConfirmReportingMethodSharedController.scala [hunk 14]
```diff
@@ -138,7 +159,7 @@ class ConfirmReportingMethodSharedController @Inject()(val manageIncomeSources:
               form = ConfirmReportingMethodForm(changeTo),
               taxYearEndYear = taxYearModel.endYear.toString,
               taxYearStartYear = taxYearModel.startYear.toString,
-              postAction = getPostAction(taxYearModel.toString, changeTo, isAgent, incomeSourceType),
+              postAction = getPostAction(taxYearModel.toString, changeTo, incomeSourceType),
```
Comment (Chrissystk) [inline app/controllers/manageBusinesses/manage/ConfirmReportingMethodSharedController.scala:162]: This is the cause of your error. You need the submitAgent when the user is an agent.

**Review by Chrissystk — CHANGES_REQUESTED** [review]

**Comment by WozCapGem** [comment]
test this please

**Review by Chrissystk — APPROVED** [review]

**Pull request merged**